Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Booking.hs

```diff
@@ -171,3 +176,29 @@ cancelBookings bookingIds now = do
         BookingUpdatedAt =. val now
       ]
     where_ $ tbl ^. BookingTId `in_` valList (toKey <$> bookingIds)
+
+findByDriverIdTripEndLatLonIfRideStatusNew :: (Transactionable m, MonadTime m) => Id Person.Person -> m (Maybe MT.LatLong)
+findByDriverIdTripEndLatLonIfRideStatusNew driverId = do
+  res <- Esq.findOne $ do
+    (_ :& _ :& _ :& bookingLocationInfo) <-
+      from $
+        table @RideT
+          `innerJoin` table @BookingT
+          `Esq.on` ( \(rideInfo :& bookingInfo) ->
+                       rideInfo ^. RideBookingId ==. bookingInfo ^. BookingTId
+                         &&. rideInfo ^. RideDriverId ==. val (toKey driverId)
+                         &&. rideInfo ^. RideStatus ==. val Ride.INPROGRESS
+                   )
+          `innerJoin` table @DriverQuoteT
+          `Esq.on` ( \(_ :& bookingInfo :& driverQuoteInfo) ->
+                       driverQuoteInfo ^. DriverQuoteId ==. bookingInfo ^. BookingQuoteId
+                         &&. bookingInfo ^. BookingStatus ==. val Booking.TRIP_ASSIGNED
+                   )
+          `innerJoin` table @BookingLocationT
+          `Esq.on` ( \(_ :& bookingInfo :& _ :& bookingLocationInfo) ->
+                       bookingInfo ^. BookingToLocationId ==. bookingLocationInfo ^. BookingLocationTId
+                   )
+    pure (bookingLocationInfo ^. BookingLocationLat, bookingLocationInfo ^. BookingLocationLon)
+  case res of
```

Comment:
can  be written as `uncurry MT.LatLong <$> res` (make sure res tuple has lat as fst and lon as snd)

-- _and don't use `x` and `y` for variable names._